hivemind/utils/grpc.py

```diff
@@ -254,10 +273,13 @@ def serialize_torch_tensor(tensor: torch.Tensor, compression_type=CompressionTyp
             size=array.shape,
             dtype=array.dtype.name,
             requires_grad=tensor.requires_grad)
-    elif compression_type == CompressionType.QUANTILE_8BIT:
+    elif compression_type == CompressionType.QUANTILE_8BIT or compression_type == CompressionType.UNIFORM_8BIT:
         assert tensor.dtype == torch.float32
 
-        quantized, lookup = quantile_encode_approx(tensor.detach(), NUM_BITS_QUANTILE_COMPRESSION)
+        if compression_type == CompressionType.QUANTILE_8BIT:
+            quantized, lookup = quantile_encode_approx(tensor.detach(), NUM_BITS_QUANTILE_COMPRESSION)
+        elif compression_type == CompressionType.UNIFORM_8BIT:
+            quantized, lookup = uint8_uniform_buckets_encode(tensor.detach(), UNIFORM_BUCKETS_STD_RANGE)
```

Comment:
I'd rather avoid nested ifs here, unless necessary

Reply:
It may be just me, but i could not infer what exactly do you propose to change

* use if-else instead of if-elif? If so, i'd request some form of assert statement to make protect from accidentally using wrong method in the future
* use functional programming magic? (e.g. `decode_func = func1 if compression_type is sth else func2`)

Reply:
I simply mean that we shouldn't overuse conditional statements inside other conditional statements, especially if they're overlapping (see L276). Writing separate function calls instead of grouping them in one block leads to dumber but simpler code, and the pieces of logic for quantile and uniform compression only overlap accidentally